core/services/scheduler.go

```diff
@@ -18,20 +18,23 @@ type Scheduler struct {
 	Recurring    *Recurring
 	OneTime      *OneTime
 	store        *store.Store
+	runManager   RunManager
 	startedMutex sync.RWMutex
 	started      bool
 }
 
 // NewScheduler initializes the Scheduler instances with both Recurring
 // and OneTime fields since jobs can contain tasks which utilize both.
-func NewScheduler(store *store.Store) *Scheduler {
+func NewScheduler(store *store.Store, runManager RunManager) *Scheduler {
```

Comment:
Again, wondering if there is any problem with having multiple copies of the RunManager.

Reply:
The RunManager has no internal state, so you can have as many copies as you like. It's just an interface for mocking purposes.

Reply:
As above `RunManager` is also an interface, which is implicitly a pointer.